src/main.ts

```diff
@@ -139,26 +140,30 @@ export function activate(context: vscode.ExtensionContext): void {
             PackageJSON.version,
             telemetryReporter);
 
-    // Create features
-    extensionFeatures = [
-        new ConsoleFeature(logger),
+    // Register commands that do not require Language client
+    commandRegistrations = [
```

Comment:
I'm not super crazy about having two different types of commands.  What if rather than implementing `IFeature`, you create a base-class called Feature (or if you can come up with a better name).  Then each subclass would use `extends` instead of `implements.  The base class would implement vscode.Disposable and would provide a `languageClient` property getter.  Then we can treat each command/feature the same with respect to storing them in a single collection in main and a single loop to dispose of them.

Reply:
I disagree.
Having a separate list, means less work to do for the `SessionManager`, which only needs to manage classes, which require management therefore due to it depending on the `languageClient`. The value of this PR is to decouple standalone and self-contained command registrations, whilst also removing boilerplate code in those classes.
`IFeature` could be called something like `ILanguageClientConsumer` to better reflect what it does.

Reply:
I'm not sure how I feel about this but I _do_ like the idea of having a base class for `LanguageClientConsumer` that sets a property `languageClient`.

```ts
export class LanguageClientConsumer {
    private _languageClient: LanguageClient;

    public get languageClient(): boolean {
        if (!_languageClient) {
              // Show warning that extension isn't done starting up
        }
        return _languageClient;
    }

    public set languageClient(value: boolean){
        this._languageClient = value;
    }
}
```

This will stop so many issues that just have the unhelpful `Unable to instantiate; language client undefined.`

Reply:
I think I'm cool with some "features" just implementing Dispose and other "features" extending this `LanguageClientConsumer`.

Did you want to make that change @bergmeister?

the if statement can just run:
```
window.showInformationMessage("PowerShell extension has not finished starting up. Please try again in a few moments.");
```

or something.

Reply:
I've just added a `LanguageClientConsumer` base class in https://github.com/PowerShell/vscode-powershell/pull/2685/commits/57b714bc5c3624d520760fe2189cbf9fe48d4c17
I couldn't remove the usage of `IFeature` yet because the `dispose()` and `setLanguageClient` methods are still being used. Trying to get rid of that would be a bit more complicated and I wouldn't like to convolute this PR too much.

Reply:
@bergmeister any reason: `LanguageClientConsumer` can't inherit from `IFeature` instead of _everything_ inheriting from `IFeature`?

Reply:
`LanguageClientConsumer` could be abstract so each thing can implement their own dispose... but then you can remove all of the:

```
 public setLanguageClient(languageClient: LanguageClient) {
        this.languageClient = languageClient;
    }
```

and have just one of those in  `LanguageClientConsumer`

Reply:
Ah let's leave those in for now. Really we should add logging ;)